lib/ibm_cloud_sdk_core/base_service.rb

```diff
@@ -80,15 +80,15 @@ def request(args)
       if args.key?(:form)
         response = conn.follow.request(
           args[:method],
-          HTTP::URI.parse(@url + args[:url]),
+          HTTP::URI.parse(@service_url + args[:url]),
```

Comment:
Somewhere in he "request" function I think there needs to be some sort of validation of the "service_url" field/variable.
At a high level, we want to allow a service instance to be constructed without a value for "service_url" to allow it to be set programmatically after the instance has been created. Then, before we actually use the "service_url" value (presumably in he request() function), we need to check to make sure it's not null/empty and raise an error if he validation fails.